### ISSUE TITLE
dependency('boost') should have a kwarg for the boost root directory

### ISSUE THREAD

**Issue by rhd** [mesonbuild/meson#249]
Hi,

Would it be possible to add a path argument to dependency() for the times when we have libs installed in non-standard places? I see that boost will look at the env variable BOOST_ROOT, but it would be nice to not have to rely on the env.

Thanks!

**Comment by germandiagogomez** [comment]
> I see that boost will look at the env variable BOOST_ROOT, but it would be nice to not have to rely on the env.

:+1 I think both use cases should be supported:

1. Use case 1: you know where they are and want to set from the meson.build file.
2. You want to honor a BOOST_ROOT variable when things do not work.

I guess the priority should be for the one set in meson.build but it should be overridable with a switch. In GNU Make this switch is `-e`.

**Comment by rhd** [comment]
I worked around this by being explicit.

```
boost_root = '/usr/local/boost_1_58_0'
boost_inc = include_directories(boost_root + '/include')
boost_lib = boost_root + '/lib'
boost_ld_flags = '-Wl,-rpath,' + boost_lib
boost_system = find_library('boost_system', dirs:boost_lib)
boost_filesystem = find_library('boost_filesystem', dirs:boost_lib)
```

**Comment by nanthony21** [comment]
The initial PR for this was closed but a follow-up PR #7210 was merged. Tis issue can probably be closed.

**Comment by eli-schwartz** [comment]
Indeed.
